crates/pathfinder/src/rpc/api.rs

```diff
@@ -1020,6 +1020,63 @@ impl RpcApi {
     //     todo!("Figure out where to take it from.")
     // }
 
+    /// Returns the contract's latest nonce.
+    ///
+    /// Not currently supported correctly as nonce's aren't implemented yet. In the mean time
+    /// returns "0x0" until starknet reaches 0.10 at which point it will return an error instead.
+    pub async fn get_nonce(&self, contract: ContractAddress) -> RpcResult<ContractNonce> {
+        // Check that contract actually exists..
+        let storage = self.storage.clone();
+        let span = tracing::Span::current();
+        let jh = tokio::task::spawn_blocking(move || {
+            let _g = span.enter();
+            let mut db = storage
+                .connection()
+                .context("Opening database connection")?;
+            let tx = db.transaction().context("Creating database transaction")?;
+
+            let exists = crate::storage::ContractsTable::exists(&tx, contract)
+                .context("Reading contract from database")?;
+
+            anyhow::Result::<_, anyhow::Error>::Ok(exists)
+        });
+        let exists = jh
+            .await
+            .context("Database read panic or shutting down")
+            .map_err(internal_server_error)?
+            .map_err(internal_server_error)?;
+        if !exists {
+            return Err(Error::from(ErrorCode::ContractNotFound));
+        }
+
+        // Check the latest known starknet version, and return "0" if its < 0.10.0
+        let version = { self.sync_state.version.read().await.clone() };
+        match version {
+            Some(version) => {
+                let version = semver::Version::parse(&version)
+                    .with_context(|| format!("Failed to parse starknet version {version}"))
+                    .map_err(internal_server_error)?;
+
+                const VERSION_0_10_0: semver::Version = semver::Version::new(0, 10, 0);
+
+                if version < VERSION_0_10_0 {
+                    Ok(ContractNonce(StarkHash::ZERO))
+                } else {
+                    Err(internal_server_error(
+                        "Not supported for StarkNet versions from 0.10.0 onwards",
+                    ))
+                }
+            }
```

Comment:
Since (AFAIK) nothing has been specced on the version, I'd prefer going with

```suggestion
            Some(version) if version.starts_with("0.9.") => Ok(ContractNonce(StarkHash::ZERO)),
            Some(version) => Err(internal_server_error(...),
```

Reply:
Okay that's reasonable. I'll ask Ariel to try confirm a specification so we can move on with our lives here and in the future.

Reply:
If you don't agree it being an opaque string, then I'd handle everything non-semver as having upgraded on the version where we supported returning zero from this instead of handling it as an error `?`.